src/Cmts.ml

```diff
@@ -270,20 +270,24 @@ let partition_after_prev_or_before_next t ~prev cmts ~next =
         let same_line_as_next l =
           next.loc_start.pos_lnum = l.loc_start.pos_lnum
         in
-        let colon_before =
-          let loc_end =
-            {prev.loc_end with pos_cnum= prev.loc_end.pos_cnum - 1}
-          in
-          let char_loc = {prev with loc_start= loc_end} in
-          let str = Source.string_at t.source char_loc in
-          String.equal str ";"
+        let infix_symbol_before =
+          let pos_cnum = prev.loc_end.pos_cnum - 1 in
+          let loc_end = {prev.loc_end with pos_cnum} in
+          match Source.position_before t.source loc_end with
+          | Some loc_start ->
+              if loc_start.pos_cnum < prev.loc_end.pos_cnum then
+                let op_loc = {prev with loc_start} in
+                let str = Source.string_at t.source op_loc in
+                String.equal str ";" || Ast.is_infix_id str
```

Comment:
Thanks for the cleanup. It does not work with keyword operators:

```
let () =
  xxxxxxxxxx
  (* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx *)
  land xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

and or-patterns:

```
let ( xxxxxxxxxx
    (* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx *)
    | xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx )
    =
  a
```

Reply:
I fixed the first case, as for the patterns, having the comment before the `|` is consistent with how it's done for other patterns kinds (see in pattern matching).